File: pattern/Two Pointers/392_Is Subsequence/Solution.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function (s, t) {
  let i = 0;
  let j = 0;
  while (i < t.length && j < s.length) {
    if (s[j] == t[i]) {
      j++;
    }
    i++;
  }
  if (j == s.length) {
    return true;
  }
  return false;
};

File: pattern/Two Pointers/392_Is Subsequence/Solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        n = len(t)
        m = len(s)
        i = 0
        j = 0
        while i < n and j < m:
            if t[i] == s[j]:
                j+=1
            i+=1
        if j == m:
            return True
        return False